src/test/csf/random.h

```diff
@@ -69,6 +69,39 @@ sample( std::size_t size, PDF pdf, Generator& g)
     std::generate(res.begin(), res.end(), [&pdf, &g]() { return pdf(g); });
     return res;
 }
+template <class T, class Generator>
+class Selector
+{
+    std::vector<T>& v_;
```

Comment:
Since this is meant to be generic code, I recommend generalizing away from `vector`.  Instead you could take begin/end iterators.  [Iterator traits](http://en.cppreference.com/w/cpp/iterator/iterator_traits) would help  and you could even static assert that the iterator has [`forward_iterator_tag`](http://en.cppreference.com/w/cpp/iterator/iterator_tags)